pkg/library/libraryinputresources/resource.go

```diff
@@ -84,11 +108,49 @@ func ResourceFromFile(location, fileTrimPrefix string) (*Resource, error) {
 
 	retFilename := strings.TrimPrefix(location, fileTrimPrefix)
 	retFilename = strings.TrimPrefix(retFilename, "/")
+	retContent := ret.(*unstructured.Unstructured)
+
+	_, apiResourceList, _, err := discoveryClient.GroupsAndMaybeResources()
```

Comment:
seems very inefficient.  We only need to get discovery once for all the files.  The GVK mapping only happens once.  in fact, you can read, then invert the matches (`map[gvk]=[]gvr`) and then pass that map into this function.

Reply:
I saw this one coming :)  I thought discovery would be cheap, since everything is vendored.

Fixed in an additional commit.